sdk/identity/azure-identity/azure/identity/aio/_credentials/chained.py

```diff
@@ -78,6 +78,7 @@ async def get_token(
                 token = await credential.get_token(*scopes, claims=claims, tenant_id=tenant_id, **kwargs)
                 _LOGGER.info("%s acquired a token from %s", self.__class__.__name__, credential.__class__.__name__)
                 self._successful_credential = credential
+                within_credential_chain.set(False)
```

Comment:
Why we set it to false after one credential?

Reply:
Since we are returning from `get_token` here, this ensures that the contextvar is set back to false. This was discovered in the unit testing when `within_credential_chain` was reading True when testing with non-chained credentials, leading to unexpected behavior.  This was because the contextvar was never set back to False.